profiles/common/modules/features/multisite_notifications/multisite_notifications_core/multisite_notifications_core.module

```diff
@@ -75,184 +75,11 @@ function multisite_notifications_core_menu() {
-  global $language;
-  $mail = $form_state['values']['mail'];
-
-  switch ($form_state['values']['op']) {
-    case t('Subscribe'):
-
-      // Create an 'anonymous' account programmatically.
-      $account = _multisite_notifications_core_create_anonymous_account($mail);
-      // Save into variables the user email.
-      variable_set('multisite_notification_core_user_email', $mail);
-      // @todo send confirmation mail.
-      $uid = $account->uid;
-      $name = $account->name;
-      $pass = $account->pass;
-
-      // Make a custom function to send mail.
-      $sent = drupal_mail(
-          // Module.
-          'multisite_notifications_core',
-          // Key.
-          'subscription',
-          // To.
-          $account->mail,
-          // Language.
-          user_preferred_language($account),
-          array($form_state['values'], $uid, $name, $pass),
-          // From.
-          _subscriptions_mail_site_mail(),
-          // If TRUE, drupal_mail() will call drupal_mail_system()->mail()
-          // to deliver the message, and store the result in $message['result'].
-          // Modules implementing hook_mail_alter() may cancel sending
-          // by setting $message['send'] to FALSE.
-          TRUE
-      );
-
-      if ($account && !empty($sent['result'])) {
-        drupal_set_message(t('You will receive a confirmation e-mail shortly containing further instructions on how to complete your subscription.'), 'status');
-      }
-
-      break;
-
-    case t('Unsubscribe'):
-
-      // Set user status to blocked, then ask for user confirmation.
-      $blocked = _multisite_notifications_core_delete_anonymous_account($mail, 'block');
-      $account = user_load_by_mail($mail);
-
-      $uid = $account->uid;
-
-      $sent = drupal_mail(
-          // Module.
-          'multisite_notifications_core',
-          // Key.
-          'unsubscription',
-          // To.
-          $account->mail,
-          // Language.
-          user_preferred_language($account),
-          array($form_state['values'], $uid),
-          // From.
-          _subscriptions_mail_site_mail(),
-          // If TRUE, drupal_mail() will call drupal_mail_system()->mail()
-          // to deliver the message, and store the result in $message['result'].
-          // Modules implementing hook_mail_alter() may cancel sending by
-          // setting $message['send'] to FALSE.
-          TRUE
-      );
-      if (!empty($sent['result'])) {
-        drupal_set_message(t('You will receive a confirmation e-mail shortly containing further instructions on how to complete your unsubscription.'), 'status');
-      }
-      else {
-        drupal_set_message(t('An error occurred when trying to send confirmation email.'), 'error');
-      }
-      break;
-
-    default:
-      break;
-
+function multisite_notifications_core_workbench_moderation_transition($node, $previous_state, $new_state) {
+  if ($new_state == 'published') {
```

Comment:
It is not a big issue but indeed on line 25 function workbench_moderation_state_published() has been used so to be consistent this line should also be changed.